### PR TITLE
Generate automatic che-theia built-in report

### PR DESCRIPTION
**Pull request by ericwill**
Generate an automatic report about the update status of che-theia built-in extensions.

See eclipse/che#17328
See eclipse/che#17014

Signed-off-by: Eric Williams <ericwill@redhat.com>

### CODE REVIEW

File: tools/automation/.ci/publish-report.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (benoitf) [inline tools/automation/.ci/publish-report.sh:1]: hi, maybe move this script in root folder 

```
<root>/.ci/publish-report.sh
```

**Comment by benoitf** [comment]
Could you please include automations in the existing yarn workspace ?
probably then drop some dependencies from this package.json

minor: seems that when running twice we've
```
 node ./lib/check-extension-updates.js 
❌ Failed to clone theia repo: GitError: fatal: destination path '/tmp/theia' already exists and is not an empty directory.
```

File: tools/automation/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true
+  },
+  "include": [
+    "src/**/*"
+  ],
+  "exclude": [
+    "node_modules"
+  ]
+}
```
Comment (benoitf) [inline tools/automation/tsconfig.json:30]: pls fix EOF

File: tools/automation/package.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
+{
+  "name": "built-in-extensions-update-automation",
+  "version": "0.0.0",
+  "private": true,
+  "description": "Scripts for updating, adding, removing, and building extensions in che-theia",
+  "main": "check-extension-updates.js",
+  "scripts": {
+    "build": "yarn run format && yarn run compile && yarn run lint",
+    "compile": "tsc --project .",
+    "format": "prettier --write **/*.ts **/**/*.ts package.json",
```
Comment (benoitf) [inline tools/automation/package.json:10]: it seems that code is not formatted b/c when I run the command there are changes

File: tools/automation/package.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    "format-check": "prettier --check **/*.ts **/**/*.ts package.json",
+    "lint": "eslint . --ext .ts",
```
Comment (benoitf) [inline tools/automation/package.json:12]: linter seems to be failing

**Comment by ericwill** [comment]
> running the command I've:
> 
> ```
> ❌ vscode-builtin-theme-dimmed error: [
>   'Failure: there is no theia built-in named *vscode-builtin-theme-dimmed*'
> ]
> ❌ vscode-builtin-merge-conflicts error: [
>   'Failure: there is no theia built-in named *vscode-builtin-merge-conflicts*'
> ]
> ❌ vscode-builtin-php error: [ 'Failure: there is no theia built-in named *vscode-builtin-php*' ]
> ❌ vscode-git error: [ 'Failure: there is no theia built-in named *vscode-git*' ]
> ```

It's because those built-ins are not part of the [Theia built-ins](https://github.com/eclipse-theia/theia/blob/8a029e2b1ffab99bdcf115b9724b167c011ec4c0/package.json#L82)

File: tools/automation/package.json (hunk 4 above)
Comment (ericwill) [reply tools/automation/package.json:12]: I am having trouble with this one:

```bash
$ eslint --cache=true --no-error-on-unmatched-pattern=true "{src,test}/**/*.{ts,tsx}"

/home/ericwill/git/che-theia/tools/automation/src/check-extension-updates.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: tools/automation/src/check-extension-updates.ts.
The file must be included in at least one of the projects provided
```

but I included it in the root `tsconfig.json`:
```js
"include": [
        "dev-packages/*/src",
        "plugins/*/src",
        "extensions/*/src",
        "tools/automation/*/src"
    ],
```

what am I missing?

**Comment by benoitf** [comment]
FYI eof-newline check is failing

**Comment by ericwill** [comment]
> FYI eof-newline check is failing

It's complaining about `/tools/automation/.gitignore` and `/tools/automation/README.md` but both files have newlines at the EOF?

**Comment by benoitf** [comment]
@ericwill : too many newlines :-)

**Comment by ericwill** [comment]
> @ericwill : too many newlines :-)

`tools/automation/README.md only has one though`

**Comment by benoitf** [comment]
it requires a newline at eof, not an empty line I would say

**Comment by benoitf** [comment]
you may want to use https://github.com/fernandrone/linelint with autofix

**Pull request merged**